app/Controllers/Http/ServiceController.ts

```diff
@@ -1,50 +1,45 @@
-// import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
-
-export default class ServicesController {}
-
-/*
-const Service = use('App/Models/Service');
-const { validateAll } = use('Validator');
-const Env = use('Env');
-const Helpers = use('Helpers');
-
-const { v4: uuid } = require('uuid');
-const path = require('path');
-const fs = require('fs-extra');
-const sanitize = require('sanitize-filename');
-
-class ServiceController {
+import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+import { schema } from '@ioc:Adonis/Core/Validator';
+import Service from 'App/Models/Service';
+import { url } from 'Config/app';
+import { v4 as uuid } from 'uuid';
+import * as fs from 'fs-extra';
+import path from 'node:path';
+import Application from '@ioc:Adonis/Core/Application';
+import sanitize from 'sanitize-filename';
+
+const createSchema = schema.create({
+  name: schema.string(),
+  recipeId: schema.string(),
+});
+
+export default class ServicesController {
   // Create a new service for user
-  async create({ request, response, auth }) {
-    try {
-      await auth.getUser();
-    } catch {
-      return response.send('Missing or invalid api token');
+  public async create({ request, response, auth }: HttpContextContract) {
+    if (!auth.user) {
+      return response.unauthorized('Missing or invalid api token');
     }
 
     // Validate user input
-    const validation = await validateAll(request.all(), {
-      name: 'required|string',
-      recipeId: 'required',
-    });
-    if (validation.fails()) {
+    let data;
+    try {
+      data = await request.validate({ schema: createSchema });
+    } catch (error) {
       return response.status(401).send({
         message: 'Invalid POST arguments',
-        messages: validation.messages(),
+        messages: error.messages,
         status: 401,
       });
     }
 
-    const data = request.all();
-
     // Get new, unused uuid
     let serviceId;
     do {
       serviceId = uuid();
     } while (
-      (await Service.query().where('serviceId', serviceId).fetch()).rows
-        .length > 0
-    ); // eslint-disable-line no-await-in-loop
+      // eslint-disable-next-line no-await-in-loop, unicorn/no-await-expression-member
+      (await Service.query().where('serviceId', serviceId)).length > 0
```

Comment:
Does this code even make sense? It is querying for a service but then the result of the call is not assigned to any variable?

Reply:
This is inside a while block. so it will run while it is > 0 (don't like it either but lets make improvements after we merge this to codebase)